cgsmiles/graph_layout.py

```diff
@@ -53,16 +89,44 @@ def assign_angles(graph):
         and the reference angle value
     """
     triplets = find_triplets(graph)
+    cycles = nx.cycle_basis(graph)
     angles = []
     for triplet in triplets:
+        # check if one is part of a ring
+        if any([triplet[0] in cycle for cycle in cycles]):
```

Comment:
`cycles` is an iterator, so this won't work well.
I suggest you make a dict of `{node_idx: cycles_containing_node_idx}` (or just the size of the smallest ring?). Actually, this isn't good enough. You need the complete triplet to be in the ring to know the angle. Either do the dict `{node: rings}`, or `{triplet: ring_sizes}`. And definitely set a max ring size, 8 sounds like a sane number.
And still you need to decide what to do with triplets that are part of multiple rings (the average maybe?). 

Finally, don't get the angles by doing the if/else tree, just say that the angle is `180 - 360/mean(len(cycle) for cycle in cycles))`